### ISSUE TITLE
Uncaught Error: Invariant Violation: EventPluginRegistry: Cannot inject two different event plugins using the same name, `TapEventPlugin`.

### ISSUE THREAD

**Issue by moroshko** [zilverline/react-tap-event-plugin#47]
When integrating multiple components that use `react-tap-event-plugin` (e.g. [`react-autosuggest`](https://github.com/moroshko/react-autosuggest)) the following error appears:
> Uncaught Error: Invariant Violation: EventPluginRegistry: Cannot inject two different event plugins using the same name, `TapEventPlugin`.

Is there a way to make sure that `react-tap-event-plugin` is instantiated only once?

**Comment by prewk** [comment]
Is there a solution for this? I'm using a `material-ui` that requires `react-tap-event-plugin`, but if I try to run `injectTapEventPlugin();` I've suddenly started getting this error as well.

**Comment by delphinefoo** [comment]
I've found a workaround for this: call injectTapEventPlugin() just once, in componentWillMount() of your top-level app component. This made the tap event plugin available in all of my child components and it was called only once.

**Comment by madjam002** [comment]
At the moment `injectTapEventPlugin` can only be called once.
@delphinefoo I wouldn't recommend putting it in `componentWillMount`, just put it at the top level of your application, just before you call `ReactDOM.render`.

**Comment by madjam002** [comment]
@prewk material-ui shouldn't be bringing in tap event plugin, you should be calling it in your application and only once. Are you sure another component isn't calling it instead as well?

See 
https://github.com/callemall/material-ui#react-tap-event-plugin

**Comment by prewk** [comment]
Yeah I wrote about it in another issue pointing here, my problem was a strange edge case with the require-new library so never mind me.

**Comment by linonetwo** [comment]
If using universal rendering, just put it over client.js, everything will be fine.

**Comment by romainPellerin** [comment]
To fix this issue only call one time var tapEventPlugin = require("react-tap-event-plugin")(); in your app or lib.
